docs/dev_guide/how_to/style_guide.md

```diff
@@ -131,6 +140,75 @@ types (e.g., `-1` for a numeric value that is otherwise expected to be positive)
 denote emptiness. For more info, here are a good [video](https://www.youtube.com/watch?v=fBYhtvY19xA&t=2317s) and 
 [post](https://homes.cs.washington.edu/~mernst/advice/nothing-is-better-than-optional.html) on this subject.
 
+### Avoid Double Brace Initialization
+
+Java supports [anonymous classes](https://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html), which are
+fine in some contexts. For example:
+
+```java
+interface MyInterface {
+  void foo();
+}
+
+class Sample {
+  void sample() {
+    MyInterface anonymousClassInstance = new MyInterface() {
+      @Override
+      public void foo() {
+        System.out.println("bar");
+      }
+    };
+  }
+}
+```
+
+Java also supports static blocks, which are executed once the first time a class is loaded. For example:
+
+```java
+class Sample2 {
+  static {
+    System.out.println("This will always be printed first, and only once.");
+  }
+  
+  void sample() {
+    System.out.println("This will be printed everytime the function is called, but always after the static block.");
+  }
+}
+```
+
+There is an antipattern which consists of initializing a collection (such as a map) with "double braces" in order to add
+elements into it. This is a combination of the above two techniques, with the first (outer) set of braces denoting the
+anonymous class, and the second (inner) set of braces denoting a static block. For example:
+
+```java
+class Sample3 {
+  void sample() {
+    // antipattern:
+    Map<String, String> map1 = new HashMap<>() {
```

Comment:
I saw/used this pattern in tests, but never realized the hidden cost of anonymous classes. 👍